cmd/nvidia-ctk-installer/container/runtime/crio/config_test.go

```diff
@@ -0,0 +1,606 @@
+
+      [crio.runtime.runtimes.nvidia-legacy]
+        monitor_path = "/usr/libexec/crio/conmon"
+        runtime_path = "/usr/bin/nvidia-container-runtime.legacy"
+        runtime_root = "/run/crun"
+        runtime_type = "oci"
+
+      [crio.runtime.runtimes.runc]
+        runtime_path = "/usr/bin/runc"
+        runtime_root = "/run/runc"
+        runtime_type = "oci"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/crio/crio.conf")
+				require.FileExists(t, configPath)
+
+				actual, err := os.ReadFile(configPath)
+				require.NoError(t, err)
+
+				// Should restore to original complex config
+				expected := `
+[crio]
+
+  [crio.image]
+    insecure_registries = ["localhost:5000"]
+    signature_policy = "/etc/crio/policy.json"
+
+  [crio.network]
+    network_dir = "/etc/cni/net.d/"
+    plugin_dirs = ["/opt/cni/bin", "/usr/libexec/cni"]
+
+  [crio.runtime]
+    conmon = "/usr/libexec/crio/conmon"
+    conmon_cgroup = "pod"
+    default_runtime = "crun"
+    selinux = true
+
+    [crio.runtime.runtimes]
+
+      [crio.runtime.runtimes.crun]
+        monitor_path = "/usr/libexec/crio/conmon"
+        runtime_path = "/usr/bin/crun"
+        runtime_root = "/run/crun"
+        runtime_type = "oci"
+
+      [crio.runtime.runtimes.runc]
+        runtime_path = "/usr/bin/runc"
+        runtime_root = "/run/runc"
+        runtime_type = "oci"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+		},
+		// Hook mode test cases
+		{
+			description: "hook mode: no existing hook",
+			containerOptions: container.Options{
+				RuntimeName: "nvidia",
+				RuntimeDir:  "/usr/bin",
+				RestartMode: "none",
+			},
+			options: Options{
+				configMode:   "hook",
+				hooksDir:     "{{ .testRoot }}/etc/crio/hooks.d",
+				hookFilename: "99-nvidia.json",
+			},
+			assertSetupPostConditions: func(t *testing.T, testRoot string) error {
+				hookPath := filepath.Join(testRoot, "etc/crio/hooks.d/99-nvidia.json")
+				require.FileExists(t, hookPath)
+
+				actual, err := os.ReadFile(hookPath)
+				require.NoError(t, err)
+
+				expectedBinary := filepath.Join("/usr/bin", config.NVIDIAContainerRuntimeHookExecutable)
+				expected := fmt.Sprintf(`{
+  "version": "1.0.0",
+  "hook": {
+    "path": "%s",
+    "args": ["nvidia-container-runtime-hook", "prestart"],
+    "env": [
+      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+    ]
+  },
+  "when": {
+    "always": true,
+    "commands": [".*"]
+  },
+  "stages": ["prestart"]
+}`, expectedBinary)
+
+				// Parse both JSONs to compare them semantically (ignoring formatting differences)
+				var expectedJSON, actualJSON interface{}
+				require.NoError(t, json.Unmarshal([]byte(expected), &expectedJSON))
+				require.NoError(t, json.Unmarshal(actual, &actualJSON))
+				require.Equal(t, expectedJSON, actualJSON)
```

Comment:
This is NOT required. We're already checking for exactly matching contents. Is there a reason to expect that these would unmarshal differently?